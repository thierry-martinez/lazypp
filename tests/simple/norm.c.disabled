// 6.10.3.3 The ## operator
// EXAMPLE
#define hash_hash # ## #
#define mkstr(a) # a
#define in_between(a) mkstr(a)
#define join(c, d) in_between(c hash_hash d)

char p[] = join(x, y); // equivalent to
                       // char p[] = "x ## y";


// 6.10.3.4 Rescanning and further replacement
// EXAMPLE
#define f(a) a*g
#define g(a) f(a)
f(2)(9)

// 6.10.3.5 Scope of macro definitions
// EXAMPLE 1
#define TABSIZE 100

int table[TABSIZE];

// EXAMPLE 2
#define max(a, b) ((a) > (b) ? (a): (b))

max(x,y)

// EXAMPLE 3

#define x 3
#define f(a) f(x * (a))
#undef x
#define x 2
#define g f
#define z z[0]
#define h g(\~{ }
#define m(a) a(w)
#define w 0,1
#define t(a) a
#define p() int
#define q(x) x
#define r(x,y) x ## y
#define str(x) # x

f(y+1) + f(f(z)) % t(t(g)(0) + t)(1);
g(x+(3,4)-w) | h 5) & m
      (f)^m(m);
p() i[q()] = { q(1), r(2,3), r(4,), r(,5), r(,) };
char c[2][6] = { str(hello), str() };
