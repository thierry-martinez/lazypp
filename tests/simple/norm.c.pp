// 6.10.3.4 Rescanning and further replacement
// EXAMPLE
2*f(9)


// 6.10.3.5 Scope of macro definitions
// EXAMPLE 1


int table[100];


// EXAMPLE 2


((x) > (y) ? (x): (y))
