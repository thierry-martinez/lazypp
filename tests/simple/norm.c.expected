// 6.10.3.3 The ## operator
// EXAMPLE

char p[] = "x ## y"; // equivalent to
                       // char p[] = "x ## y";

// 6.10.3.4 Rescanning and further replacement
// EXAMPLE
2*f(9)


// 6.10.3.5 Scope of macro definitions
// EXAMPLE 1


int table[100];


// EXAMPLE 2


((x) > (y) ? (x): (y))


// EXAMPLE 3

f(2 * (y+1)) + f(2 * (f(2 * (z[0])))) % f(2 * (0)) + t(1);
f(2 * (2+(3,4)-0,1)) | f(2 * (\~{ } 5)) & f(2 * (0,1))^m(0,1);
int i[] = { 1, 23, 4, 5, };
char c[2][6] = { "hello", "" };